instrgen/driver/main.go

```diff
@@ -0,0 +1,168 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"log"
+	"os"
+
+	alib "go.opentelemetry.io/contrib/instrgen/lib"
+)
+
+func usage() error {
+	fmt.Println("\nusage driver --command [path to go project] [package pattern]")
+	fmt.Println("\tcommand:")
+	fmt.Println("\t\tinject                                 (injects open telemetry calls into project code)")
+	fmt.Println("\t\tinject-dump-ir                         (injects open telemetry calls into project code and intermediate passes)")
+	fmt.Println("\t\tprune                                  (prune open telemetry calls")
+	fmt.Println("\t\tdumpcfg                                (dumps control flow graph)")
+	fmt.Println("\t\trootfunctions                          (dumps root functions)")
+	return nil
+}
+
+func makeAnalysis(projectPath string, packagePattern string, debug bool) *alib.PackageAnalysis {
+	var rootFunctions []alib.FuncDescriptor
+
+	interfaces := alib.FindInterfaces(projectPath, packagePattern)
+	rootFunctions = append(rootFunctions, alib.FindRootFunctions(projectPath, packagePattern, "AutotelEntryPoint")...)
+	funcDecls := alib.FindFuncDecls(projectPath, packagePattern, interfaces)
+	backwardCallGraph := alib.BuildCallGraph(projectPath, packagePattern, funcDecls, interfaces)
+	fmt.Println("\n\tchild parent")
+	for k, v := range backwardCallGraph {
+		fmt.Print("\n\t", k)
+		fmt.Print(" ", v)
+	}
+	fmt.Println("")
```

Comment:
Is this useful for user ?

Reply:
This is useful for debugging purposes. I left just for now, but of course we can get rid of this printfs.